From a5e33b2dbd8301b8f772bad5db4019587b2034a0 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Tue, 1 Oct 2024 06:42:57 +0000
Subject: [PATCH] scanner: Parse _Alignas and discard it

Scanner should not choke on _Alignas as it is part of C11.
https://en.cppreference.com/w/c/language/_Alignas

Fixes: #520
---
 giscanner/scannerlexer.l | 1 +
 1 file changed, 1 insertion(+)

diff --git a/giscanner/scannerlexer.l b/giscanner/scannerlexer.l
index 19aa3766e..38e21534e 100644
--- a/giscanner/scannerlexer.l
+++ b/giscanner/scannerlexer.l
@@ -166,6 +166,7 @@ stringtext				([^\\\"])|(\\.)
 "__asm__" 	        	        { if (!parse_ignored_macro()) REJECT; }
 "__attribute__" 		        { if (!parse_ignored_macro()) REJECT; }
 "__attribute" 		                { if (!parse_ignored_macro()) REJECT; }
+"_Alignas" 		                { if (!parse_ignored_macro()) REJECT; }
 "__const"                               { return CONST; }
 "__extension__"                         { return EXTENSION; }
 "__forceinline"                         { return INLINE; }
-- 
GitLab

